internal/status/route_status.go

```diff
@@ -580,13 +597,21 @@ func deleteRouteObject(option UpdateOptions, key string, isVSDelete bool, retryN
 
 	utils.AviLog.Infof("key: %s, deleting hostnames %v from Route status %s/%s", key, option.ServiceMetadata.HostNames, option.ServiceMetadata.Namespace, option.ServiceMetadata.IngressName)
 	svcMdataHostname := option.ServiceMetadata.HostNames[0]
+	routeHost := mRoute.Spec.Host
+	if routeHost == "" && mRoute.Spec.Subdomain != "" && lib.GetDomain() != "" {
```

Comment:
we can have a utility function returning hostname instead of  this if block since it is used in multiple palces

Reply:
Done